### DISCUSSION TITLE
How do I make run mode commands open in the shell automatically?

### DISCUSSION THREAD

**Discussion by retroflexivity** [davatorium/rofi#2038] [Q&A] [answered]
In my config, I have `run-command: "{cmd}"` and `run-shell-command: "{terminal} --hold {cmd}"`. As I understand it, normally, the first one is executed on `enter` and the second one on `shift-enter`, independently of whether the selected entry is in fact a shell app or a desktop app. 

What I would like is for the `enter` key to open desktop apps directly, and shell apps in the terminal, that is, execute `run-command` for drun entries and `run-shell-command` for run entries.

Is it possible to achieve such behaviour in rofi?

**Comment by DaveDavenport** [comment, marked as answer]
For drun if rofi uses 'run-command' or 'run-shell-command'  depends on  what the desktop file indicate. 
For run it is up to the user. 

There is currently no way to have  to change settings like these that only apply for a single mode.  (unless you run two separate sessions _

**Reply by DaveDavenport** [reply]
> independently of whether the selected entry is in fact a shell app or a desktop app.

For 'run' mode that just looks at executables in `$PATH`, how could rofi know if it is a 'shell app' or not?
So the solution is `drun` that has this meta information available in `.desktop` files.

**Reply by retroflexivity** [reply]
> There is currently no way to have to change settings like these that only apply for a single mode. (unless you run two separate sessions _

welp, seems like this answers my question. thank you
